### PR TITLE
[uio 33]related products initial

### PR DESCRIPTION
**Pull request by moskdev**

### CODE REVIEW

File: src/app/components/ProductDetails/ProductDetails.js [hunk 1]
```diff
@@ -1,16 +1,62 @@
...
-const ProductDetails = props => {
-  const { id, products } = props;
+class ProductDetails extends Component {
+  constructor(props) {
+    super(props);
+    this.state = { relatedProducts: [] };
+  }
 
-  const item = products.find(el => el._id === id);
+  componentDidMount() {
```
Comment (Hollister009) [inline src/app/components/ProductDetails/ProductDetails.js]: Consider not to use this approach very often. It's a bad practice

File: src/app/components/RelatedProducts/RelatedProducts.js [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+      </div>
+      <Carousel data={products} extended>
+        {list}
+      </Carousel>
+    </section>
+  );
+};
+
+RelatedProducts.propTypes = {
+  products: PropTypes.arrayOf(PropTypes.object),
```
Comment (Hollister009) [inline src/app/components/RelatedProducts/RelatedProducts.js]: Not the best way to describe products.
I have created actual file which describes single product shape in _types_ folder

Comment (Hollister009) [reply src/app/components/RelatedProducts/RelatedProducts.js]: Also I would use isRequired instead of empty default arrays

Comment (Hollister009) [reply src/app/components/RelatedProducts/RelatedProducts.js]: @moskdev https://medium.freecodecamp.org/react-pattern-centralized-proptypes-f981ff672f3b consider this manual

File: src/app/components/ProductDetails/ProductDetails.js [hunk 3]
```diff
@@ -1,16 +1,62 @@
...
+    this.forceUpdate();
```
Comment (andriytereschuk) [inline src/app/components/ProductDetails/ProductDetails.js]: what is going on?
`this.forceUpdate();`
`window.scrollTo(0, 0);`

File: src/app/components/ProductDetails/ProductDetails.js [hunk 4]
```diff
@@ -1,16 +1,62 @@
...
 import ProductDescription from '../ProductDescription/ProductDescription';
+import RelatedProducts from '../RelatedProducts/RelatedProducts';
+import HttpService from '../../../utils/http.service';
 
-const ProductDetails = props => {
-  const { id, products } = props;
+class ProductDetails extends Component {
+  constructor(props) {
+    super(props);
+    this.state = { relatedProducts: [] };
```
Comment (andriytereschuk) [inline src/app/components/ProductDetails/ProductDetails.js]: relatedProducts have to be stored inside `relatedProducts` component

File: src/app/components/ProductDetails/ProductDetails.js [hunk 5]
```diff
@@ -1,16 +1,62 @@
...
-      <ProductDescription item={item} />
-    </div>
-  );
-};
+  componentDidUpdate() {
+    const { products, id } = this.props;
+    const { relatedProducts } = this.state;
+    const item = products.find(el => el._id === id);
+
+    if (item && !relatedProducts.length) {
```
Comment (andriytereschuk) [inline src/app/components/ProductDetails/ProductDetails.js]: move the stuff that is concerned related products to `relatedProducts` component

**Comment by Hollister009** [comment]
Done!

File: src/app/components/ProductDetails/ProductDetails.js [hunk 6]
```diff
@@ -1,16 +1,62 @@
...
+      const params = {
```
Comment (andriytereschuk) [inline src/app/components/ProductDetails/ProductDetails.js]: would be better to pass only needed filter params

and override default params

I suggest using some service class to handle this

**Pull request merged**